packages/integration-tests/src/test-project-setup/test_project_base.ts

```diff
@@ -90,4 +95,18 @@ export abstract class TestProjectBase {
   async getUpdates(): Promise<TestProjectUpdate[]> {
     return [];
   }
+
+  /**
+   * Verify client config file is generated with the provided directory and format
+   */
+  async assertPostDeployment(dir?: string, format?: ClientConfigFormat) {
+    const clientConfigStats = await fsp.stat(
+      await getClientConfigPath(dir ?? this.projectDirPath, format)
```

Comment:
do you have latest `main` in this PR?
I'd expect this fails on `json-mobile` https://github.com/aws-amplify/amplify-backend/blob/14274f43776d8233faaa69d870dcf4fa9e8e0a3b/packages/client-config/src/client-config-types/client_config.ts#L19 .

My other thought is that I'd keep `assertPostDeployment` as it was before because it's part of deploy-assert mechanism (and assumes default flags deployment runs with, has nothing to do with extra config generation).

For probing client config configurations I'd just create separate set of assertions.

Reply:
Yup I saw it failed on `json-mobile` before getting latest `main`, now it is passing locally.

I see, I was thinking to reuse `assertPostDeployment` since it was checking for existence of `amplifyconfiguration.json` in project directory. With no parameters this acts the same way as before with an added benefit of not having to change the default format if we decide to change it in the future.

Reply:
That looks like accidental duplication per https://gtramontina.com/posts/do-repeat-yourself/ .
these assertions are expected to evolve independently, let's not couple them.

Reply:
Ahh.. I totally missed `getClientConfigPath` . (that's re `json-mobile`).

As for accidental duplication I think that:
1. It would be fine to define `assertClientConfigExists` which has logic in current form.
2. Use `assertClientConfigExists` in `assertPostDeployment`

This would address both problems, i.e. it would eliminate duplication AND didn't bend the "post deployent assertion" abstraction to different purpose.

Reply:
looks good, now thanks!

Reply:
Yeah used `getClientConfigPath` so everything is aligned (case of `json-mobile` -> `json` and defaults) and we would only need to change one spot.

Done in [971d26e](https://github.com/aws-amplify/amplify-backend/pull/682/commits/971d26e3eeb7332a00291e527bf4232db2f7ac09)